src/Workspaces/Remote/ServiceHub/Services/RemoteHostService.cs

```diff
@@ -219,5 +224,22 @@ private static AbstractPersistentStorageService GetPersistentStorageService()
             var persistentStorageService = workspace.Services.GetService<IPersistentStorageService>() as AbstractPersistentStorageService;
             return persistentStorageService;
         }
+
+        [DllImport("kernel32.dll", SetLastError = true)]
+        private static extern IntPtr AddDllDirectory(string directory);
+
+        private static void SetNativeDllSearchDirectories()
+        {
+            // Set LoadLibrary search directory to %VSINSTALLDIR%\Common7\IDE so that the compiler 
+            // can P/Invoke to Microsoft.DiaSymReader.Native when emitting Windows PDBs.
+
+            Contract.Assert(string.Equals(Path.GetDirectoryName(AppDomain.CurrentDomain.BaseDirectory), "IDE", StringComparison.OrdinalIgnoreCase));
```

Comment:
ServiceHub sets the appdomain base directory according to your `*.servicehub.service.json` file.